core/src/main/scala/sttp/openai/OpenAISyncClient.scala

```diff
@@ -0,0 +1,447 @@
+package sttp.openai
+
+import sttp.client4.{DefaultSyncBackend, DeserializationException, HttpError, Request, Response, ResponseException, SyncBackend}
+import sttp.openai.OpenAIErrors.OpenAIError
+import sttp.openai.OpenAIErrors.OpenAIError._
+import sttp.openai.json.SnakePickle
+import sttp.openai.requests.audio.AudioResponseData.AudioResponse
+import sttp.openai.requests.audio.transcriptions.TranscriptionConfig
+import sttp.openai.requests.audio.translations.TranslationConfig
+import sttp.openai.requests.audio.{AudioResponseData, RecognitionModel}
+import sttp.openai.requests.completions.CompletionsRequestBody.CompletionsBody
+import sttp.openai.requests.completions.CompletionsResponseData.CompletionsResponse
+import sttp.openai.requests.completions.chat.ChatRequestBody.ChatBody
+import sttp.openai.requests.completions.chat.ChatRequestResponseData
+import sttp.openai.requests.completions.chat.ChatRequestResponseData.ChatResponse
+import sttp.openai.requests.completions.edit.EditRequestBody.EditBody
+import sttp.openai.requests.completions.edit.EditRequestResponseData
+import sttp.openai.requests.completions.edit.EditRequestResponseData.EditResponse
+import sttp.openai.requests.embeddings.EmbeddingsRequestBody.EmbeddingsBody
+import sttp.openai.requests.embeddings.EmbeddingsResponseBody
+import sttp.openai.requests.embeddings.EmbeddingsResponseBody.EmbeddingResponse
+import sttp.openai.requests.files.FilesResponseData
+import sttp.openai.requests.files.FilesResponseData.{DeletedFileData, FileData, FilesResponse}
+import sttp.openai.requests.finetunes.FineTunesResponseData.{
+  DeleteFineTuneModelResponse,
+  FineTuneEventsResponse,
+  FineTuneResponse,
+  GetFineTunesResponse
+}
+import sttp.openai.requests.finetunes.{FineTunesRequestBody, FineTunesResponseData}
+import sttp.openai.requests.images.ImageResponseData.ImageResponse
+import sttp.openai.requests.images.creation.ImageCreationRequestBody.ImageCreationBody
+import sttp.openai.requests.images.edit.ImageEditsConfig
+import sttp.openai.requests.images.variations.ImageVariationsConfig
+import sttp.openai.requests.models.ModelsResponseData.{ModelData, ModelsResponse}
+import sttp.openai.requests.moderations.ModerationsRequestBody.ModerationsBody
+import sttp.openai.requests.moderations.ModerationsResponseData
+import sttp.openai.requests.moderations.ModerationsResponseData.ModerationData
+
+import java.io.File
+
+class OpenAISyncClient private (authToken: String, backend: SyncBackend) {
+
+  private val openAI = new OpenAI(authToken)
+
+  /** Lists the currently available models, and provides basic information about each one such as the owner and availability.
+    *
+    * [[https://platform.openai.com/docs/api-reference/models]]
+    */
+  def getModels: Either[OpenAIError, ModelsResponse] =
```

Comment:
I think the sync client should use exceptions instead of `Either`s - providing a quick gateway to openai; if somebody wants more control over error cases, there's the `F`-backend which exposes eithers